### PR TITLE
Move DNS lookup metric outside of .NET semconv

### LINKED ISSUE #404: Define general metric for DNS lookup duration
We're adding .NET-specific one in #283, but it's generally useful to have it for other languages and platforms.

This would also include bringing in some if the ECS [DNS attributes](https://www.elastic.co/guide/en/ecs/current/ecs-dns.html).

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #404

## Changes

This PR moved DNS lookup duration outside of .NET semantic conventions. 

It removes MD generation from .NET metric definition as it's frozen in time and should not get any updates or changes on the common metric.

It helps to untie .NET semconv stability from the unstable OTel parts it depends upon and helps with #781 

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

**Comment by lmolkova** [comment]
@open-telemetry/dotnet-approvers what do you think about this approach? We have a few more places where .NET either uses experimental attributes on stable metrics or declares common-purpose metrics like `http.connection.duration`.

The proposal here is to move such things away into general semconv as experimental and, as a precaution, remove table auto-generation from .NET preventing possible breaking changes on .NET docs.

The context: we're explicitly annotating stability on attributes and metrics and realized we have some conflicts. Oops

**Comment by vishweshbankwar** [comment]
> @open-telemetry/dotnet-approvers what do you think about this approach? We have a few more places where .NET either uses experimental attributes on stable metrics or declares common-purpose metrics like `http.connection.duration`.
> 
> The proposal here is to move such things away into general semconv as experimental and, as a precaution, remove table auto-generation from .NET preventing possible breaking changes on .NET docs.
> 
> The context: we're explicitly annotating stability on attributes and metrics and realized we have some conflicts. Oops

@lmolkova The change looks good to me. But I think we should make the note visible on .NET side explaining what does **stable** mean from .NET's perspective?.

**Comment by noahfalk** [comment]
This seems fine to me. It does raise a larger question... if there was concern about unintended breaking changes on the .NET docs for this metric, does that same risk exist for other metrics?

> But I think we should make the note visible on .NET side explaining what does stable mean from .NET's perspective?

I'm not sure we've ever officially defined it, but I interpret it to mean we shipped the behavior and we'd treat it similar to any other .NET API behavior in terms of preserving back-compat. Namely that means we'd try hard not to break it without very good reason and if we did introduce a breaking change it would be [officially documented](https://learn.microsoft.com/en-us/dotnet/core/compatibility/9.0).

File: docs/dotnet/dotnet-dns-metrics.md [hunk 1]
```diff
@@ -19,19 +19,17 @@ This article defines semantic conventions for DNS metrics emitted by .NET.
 
 ### Metric: `dns.lookup.duration`
 
-this metric SHOULD be specified with
+This metric SHOULD be specified with
 [`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advisory-parameters)
 of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10 ]`.
 
-<!-- semconv metric.dotnet.dns.lookup.duration(metric_table) -->
+<!-- Tables in this document are not auto-generated and are intentionally frozen in time. From the .NET perspective this metric and its attributes are stable till the next major version. They are still experimental in the OpenTelemetry. -->
```
Comment (antonfirsov) [inline docs/dotnet/dotnet-dns-metrics.md:26]: I assume the reason the paragraph is not simply replaced with a reference to the common metric (like with `http.client.duration`) is the fact that the OTel definition is unstable? Is the plan to change this later?

**Comment by lmolkova** [comment]
> if there was concern about unintended breaking changes on the .NET docs for this metric, does that same risk exist for other metrics?

@noahfalk not for stable ones, the problem that just became obvious that .NET stable document uses metrics and attributes that are not defined as stable.

I.e. there are no breaking changes coming to `http.client.request.duration` or any stable attributes, but it is possible that `dns.question.name` attribute is renamed in the future because from otel standpoint it's experimental. 

There are other metrics I want to extract from .NET in follow-up PRs - everything that starts with [`http.*`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/dotnet/dotnet-http-metrics.md). Since they are not in the .NET namespace, there is always a chance they will be used by other semconvs and applications.

I'd like to do the same trick I've done  in this PR with them.

Assuming we do this, I think we can handle changes in the following way
1. if OTel stabilizes `dns.question.name` and corresponding metric without any incompatible changes
  -  then .NET semconv can be updated to reference existing metric instead of copy-paste.
2. if any breaking changes happen on the OTel side, .NET has a choice:
  - to stay with non-standard metric/attribute - then we'll keep copy-paste for .NET
  - deprecate/break in the next major version and follow OTel-defined stable metric/attribute

**Comment by joaopgrassi** [comment]
> * if OTel stabilizes `dns.question.name` and corresponding metric without any incompatible changes
>   
>   * then .NET semconv can be updated to reference existing metric instead of copy-paste.
> 
> * if any breaking changes happen on the OTel side, .NET has a choice:
>   
>   * to stay with non-standard metric/attribute - then we'll keep copy-paste for .NET
>   * deprecate/break in the next major version and follow OTel-defined stable metric/attribute

This plan makes sense to me.

**Comment by lmolkova** [comment]
@open-telemetry/semconv-dotnet-approver any concerns here?

**Review by antonfirsov — APPROVED** [review]

File: model/registry/dns.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    brief: >
+      This document defines the shared attributes used to report a DNS query.
+    attributes:
+      - id: question.name
+        type: string
+        stability: experimental
+        brief: The name being queried.
+        examples: ["www.example.com", "opentelemetry.io"]
+        note: >
+          The name being queried.
```
Comment (trisch-me) [inline model/registry/dns.yaml:14]: I think note shouldn't repeat the `brief` content

File: .chloggen/785.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+change_type: enhancement
+component: dotnet
```
Comment (trisch-me) [inline .chloggen/785.yaml:2]: shouldn't it be `dotnet, dns`?

**Review by trisch-me — APPROVED** [review]
lgtm, left few comments

File: model/registry/dns.yaml (hunk 2 above)
Comment (lmolkova) [reply model/registry/dns.yaml]: great catch!

File: .chloggen/785.yaml (hunk 3 above)
Comment (lmolkova) [reply .chloggen/785.yaml]: I believe we don't support multiple components in the changelog, so I changed it to `dns`, nothing has changed from the .NET perspective here anyway

**Review by joaopgrassi — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Pull request merged**